internal/codegen/stencil_test.go

```diff
@@ -121,3 +116,47 @@ func ExampleStencil_PostRun() {
 	// Output:
 	// hello
 }
+
+func TestStencilPostRunError(t *testing.T) {
+	const (
+		name    = "TestStencilPostRunError"
+		command = "invalidPostRunCommand"
+	)
+
+	var (
+		ctx              = context.Background()
+		logger           = logrus.New()
+		manifestContents = fmt.Sprintf("name: %s\npostRunCommand:\n- command: %s\n", name, command)
+		errMsg           = fmt.Sprintf("failed to run post run command for module %s and command %s", name, command)
+	)
+
+	fs := createFakeModuleFSWithManifest(t, manifestContents)
+	st := NewStencil(
+		&configuration.ServiceManifest{Name: name},
+		[]*modules.Module{modules.NewWithFS(ctx, name, fs)},
+		logger,
+	)
+
+	assert.ErrorContains(t, st.PostRun(ctx, logger), errMsg)
+}
+
+// createFakeModuleFSWithManifest creates an in-memory filesystem with a single
+// file named "manifest.yaml" containing the provided manifest contents. This
+// is useful for testing purposes where a mock filesystem is needed.
+//
+// Parameters:
+//   - t: The testing object used for assertions.
+//   - manifestContents: A string representing the contents to be written to the manifest file.
+//
+// Returns:
+//   - A billy.Filesystem representing the in-memory filesystem with the manifest file.
+func createFakeModuleFSWithManifest(t *testing.T, manifestContents string) billy.Filesystem {
+	fs := memfs.New()
```

Comment:
suggestion: denote to the testing context that this is a [helper](https://pkg.go.dev/testing#T.Helper). This is so that when the file/line of the assertion error is printed out, it's not printing the information of the helper function.
```suggestion
func createFakeModuleFSWithManifest(t *testing.T, manifestContents string) billy.Filesystem {
	t.Helper()
	fs := memfs.New()
```